app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java

```diff
@@ -349,6 +349,7 @@ private void initFragments() {
     private void showFragment(Fragment fragment, String tag, Fragment otherFragment) {
         FragmentTransaction transaction = getChildFragmentManager().beginTransaction();
         if (fragment.isAdded() && otherFragment != null) {
+            removeFragment(otherFragment);
```

Comment:
Hey @Pratham2305 , thanks a lot for the PR. It works as intended. Code looks nice in general. However, I couldn't understand why this line was needed to achieve this task. Can you please explain?

Reply:
While debugging I have noticed that after recreating, setfragment method gets called to set the ContributionListFragment but the media fragment is not removed which is causing this problem, as the removeFragment() method is only called on back pressed. So I replaced the removeFragment() to the setFragment method as after pressing the back button also this same method gets called.

Reply:
The issue can also be solved by simply calling the show tab method and disabling the toolbar back button on return similar to that I have done in case of explore or bookmark fragment, but I thought this would be the better solution in this case.

Reply:
Everytime removing the otherfragment (means MediaDetailsPagerFragment) doesn't seem like an efficient solution to me. Also in the next line we hide a fragment which is already removed, which makes no sense. 

Also, semantically it is better to remove mediaDetailPagerFragment on back button is clicked, so I think the line 668 should not be removed. When it comes to the solution of the problem, what about removing mediaDetailPagerFragment when the contributionFragment is removed or destroyed?

Reply:
I agree with you,  removing and hiding the same fragment makes no sense, sorry about the previous changes.
I think the issue is after recreating, So I have added a condition in onCreateView() to check if mediaDetailPagerFragment is null or not and if not null then simply removed the mediaDetailPagerFragment before setting the contributionListFragment.

Reply:
Sounds like a plan :) Does this still work to solve the issue of theme change though?

Reply:
Yes, I have tested it and it is working as expected.